File: 05_Check for anagram/Ideal Solutions/checkAnagram.java
import java.util.*;
import java.lang.*;
import java.io.*;



class Main{
	//This function checks if the two arrays are equal or not.
	public static boolean equalArrays(char[] arr1, char[] arr2){
		if (arr1.length != arr2.length) return false;
		for (int i=0;i<arr1.length;i++){
			if (arr1[i] != arr2[i]) return false;
		}
		return true;
	}
	//This function checks if the two strings are anagrams of each other.
	public static boolean checkAnagram(String a,String b){
		char[] arr1 = a.toCharArray();
		char[] arr2 = b.toCharArray();
		Arrays.sort(arr1);
		Arrays.sort(arr2);
		return equalArrays(arr1,arr2);
	}
	//Main Function
	public static void main(String[] args){
		int n = 4;
		String a = "abcd";
		int m = 4;
		String b = "adcb";
		if (checkAnagram(a,b)) System.out.println("Yes");
		else{
			System.out.println("No");
		}
	}
}

File: 05_Check for anagram/Ideal Solutions/checkAnagram.cpp
#include<bits/stdc++.h>
using namespace std;
bool checkAnagram(string x, string y, int n, int m) {
	// Function to check if two strings are anagram or not

	// Using two arrays of contant size to count the frequency of each character present in the string.
	int freq1[27], freq2[27];
	memset(freq1, 0, sizeof(freq1)); // Setting initial values in the freq1 array to be 0
	memset(freq2, 0, sizeof(freq2)); // Setting initial values in the freq2 array to be 0
	for (int i = 0; i < n; i++)
	{
		int c = (int)x[i] - 96;
		freq1[c]++;
	}
	for (int i = 0; i < m; i++)
	{
		int c = (int)y[i] - 96;
		freq2[c]++;
	}

	// Checking if the frequency of each character is same or not
	for (int i = 1; i <= 26; i++)
		if (freq1[i] != freq2[i])
			return false;
	return true;
}
int main() {
	int N = 4, M = 4; // Length of the strings
	string X = "abcd";
	string Y = "dacb";
	if (checkAnagram(X, Y, N, M))
		cout << "Yes" << endl;
	else
		cout << "No" << endl;
	return 0;
}
